pkg/storegateway/stats.go

```diff
@@ -76,8 +77,17 @@ type queryStats struct {
 	streamingSeriesAmbientTime time.Duration
 }
 
+// blockSourceAndLevel encapsulate a block's thanos source and compaction level
+type blockSourceAndLevel struct {
+	source string
+	level  int
+}
+
 func (s queryStats) merge(o *queryStats) *queryStats {
 	s.blocksQueried += o.blocksQueried
+	for sl, count := range o.blocksQueriedBySourceAndLevel {
+		s.blocksQueriedBySourceAndLevel[sl] += count
```

Comment:
should we account for a nil `s.blocksQueriedBySourceAndLevel`?

Reply:
I added a `new` method to initialize that field. It would only be nil if someone didn't use the new method, in which case we'd probably want a panic so that we can learn about the bug. Seem good?

Reply:
sounds good. thanks